apps/registry/codemods/eslint/biome/migrate-rules/src/functions.ts

```diff
@@ -0,0 +1,291 @@
+import type { DataAPI } from '@codemod-com/filemod';
+import { isNeitherNullNorUndefined } from '@codemod-com/utilities/functions/validationMethods';
+import type { Input } from 'valibot';
+import type {
+	Configuration as BiomeConfig,
+	Rules as BiomeLinterRules,
+} from '../types/biome.js';
+import type { OptionsDefinition as PrettierConfig } from '../types/prettier.js';
+import type { packageJsonSchema } from './schemas.js';
+import type { Dependencies, RuleValue } from './types.js';
+
+export function replaceKeys(
+	obj: Record<string, unknown>,
+	replacements: Record<string, string>,
+): Record<string, unknown> {
+	// Just in case
+	if (typeof obj !== 'object' || obj === null) {
+		return obj;
+	}
+
+	const newObj = Object.assign({}, obj);
```

Comment:
this creates shallow copy

Reply:
should i do `JSON.parse(JSON.stringify(obj))`?